kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx

```diff
@@ -398,6 +398,18 @@ export function ReportSettings(props: ReportSettingProps) {
     }
   };
 
+  const setInContextDefaultConfiguration = () => {
+    const url = window.location.href;
+    const id = parseInContextUrl(url, 'id');
+    if (url.includes('dashboard')) {
```

Comment:
This is a corner case, but if I have a visualization which title contains `dashboard`, would the source be identified as dashboard? and would that break Reporting?

Reply:
good catch

Reply:
No- the url constructed doesn't use the name of the report source but the id. `dashboard` or `visualization` is included in the url depending on which page we came from, a dashboard or url.

Reply:
got it, thanks